adapter/interface.go

```diff
@@ -0,0 +1,56 @@
+// Package adapter allows this CLI to switch between clients provided by fn_go and oci-go-sdk
+// https://en.wikipedia.org/wiki/Adapter_pattern
+package adapter
+
+import (
+	"fmt"
+)
+
+type Provider interface {
+	APIClient() APIClient
+	VersionClient() VersionClient
+	FunctionInvokeClient() FunctionInvokeClient
+}
+
+type APIClient interface {
+	FnClient() FnClient
+	AppClient() AppClient
+	TriggerClient() TriggerClient
+}
+
+type FnClient interface {
+	CreateFn(fn *Fn) (*Fn, error)
+	UpdateFn(fn *Fn) (*Fn, error)
+	GetFn(appID string, fnName string) (*Fn, error)
+	ListFn(appID string, limit int64) error
+	DeleteFn(fnID string) error
+}
+
+type AppClient interface {
+	CreateApp(app *App) (*App, error)
+	GetApp(appName string) (*App, error)
+	UpdateApp(app *App) (*App, error)
+	ListApp(limit int64) ([]*App, error)
+	DeleteApp(appID string) error
+}
+
+type TriggerClient interface {
```

Comment:
We do know what methods it will have. The same ones used to make manage Triggers in Fn CLI now. This interface will have 2 implementations. An OSS one, that will send calls similar to how we do things now. An OCI one that will be a stub and return an appropriate error response, since we don't support Trigger in OCI Functions.